src/components/Product.js

```diff
@@ -23,39 +25,68 @@ function Product() {
     setNoOfCartItems(isProductUnique.length);
   }, [isProductUnique.length, setNoOfCartItems]);
 
-  function increase() {
-  }
-
-  function decrease() {
-    
-  }
-
   function addProducTtoCart() {
+    if (cartProduct) {
+      setCartProduct((product) =>
+        product?.map((item) =>
+          products[url ? url - 1 : 1].id === item.id
+            ? { ...item, quantity: item.quantity + 1 }
+            : item
+        )
+      );
+    }
     setCartProduct((prev) => [
       ...prev,
       {
         id: products[url ? url - 1 : 1].id,
         image: products[url ? url - 1 : 1].image,
         title: products[url ? url - 1 : 1].title,
         price: products[url ? url - 1 : 1].price,
-        color: pickColor,
+        quantity: products[url ? url - 1 : 1].quantity,
+        color: products[url ? url - 1 : 1].color,
       },
     ]);
     setNoOfCartItems(isProductUnique.length);
   }
 
-  console.log(cartProduct);
-  console.log(isProductUnique);
+  function decrease() {
+    if (products[url ? url - 1 : 1].quantity > 1) {
```

Comment:
this line "url ? url - 1 : 1" is both repeated alot and not easy to understand. Please fix it.